charts/argo-cd/templates/argocd-repo-server/deployment.yaml

```diff
@@ -104,6 +104,9 @@ spec:
     {{- if .Values.repoServer.affinity }}
       affinity:
 {{- toYaml .Values.repoServer.affinity | nindent 8 }}
+    {{- end }}
+    {{- if .Values.repoServer.serviceAccount.create }}
```

Comment:
Could you please remove this conditional so we can support the three use cases:
- create: false, service account name unset: template defaults to serviceAccountName: "default"
- create: false, service account name set: template uses specific service account name (common use case for externally created service accounts)
- create: true: service account name set: new service account created